test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/lang/ClassTest.scala

```diff
@@ -100,13 +112,26 @@ class ClassTest {
 
     assertTrue(classOf[Object].isAssignableFrom(classOf[String]))
     assertTrue(classOf[Seq[_]].isAssignableFrom(classOf[List[_]]))
+    assertTrue(classOf[List[_]].isAssignableFrom(classOf[::[_]]))
+    assertTrue(classOf[Seq[_]].isAssignableFrom(classOf[::[_]]))
     assertTrue(classOf[Object].isAssignableFrom(classOf[Array[String]]))
     assertTrue(classOf[Array[Seq[_]]].isAssignableFrom(classOf[Array[List[_]]]))
 
     assertFalse(classOf[String].isAssignableFrom(classOf[Object]))
     assertFalse(classOf[List[_]].isAssignableFrom(classOf[Seq[_]]))
+    assertFalse(classOf[Option[_]].isAssignableFrom(classOf[::[_]]))
+    assertFalse(classOf[Set[_]].isAssignableFrom(classOf[::[_]]))
     assertFalse(classOf[Array[String]].isAssignableFrom(classOf[Object]))
     assertFalse(classOf[Array[List[_]]].isAssignableFrom(classOf[Array[Seq[_]]]))
+
+    for {
+      cls <- BoxedClassOfs
+      if cls != classOf[java.lang.Void]
+    } {
+      assertTrue(
+          s"classOf[Comparable[_]].isAssignableFrom($cls) should be true",
+          classOf[Comparable[_]].isAssignableFrom(cls))
+    }
```

Comment:
Maybe add a comment that boxed classes extend `Comparable` and since they are hijacked, it makes sense to test that in addition to subclassing in general?